### PR TITLE
[GitHub] Index markdown files only

### PR DESCRIPTION
**Pull request by abhishekjoshi-crest**
## Closes https://github.com/elastic/connectors-python/issues/1350

As discussed on slack, In GitHub we are indexing markdown files only.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/sources/github.py [hunk 1]
```diff
@@ -1287,39 +1288,32 @@ async def _fetch_files(self, repo_name, default_branch):
             )
 
             for repo_object in file_tree.get("tree", []):
-                last_commit_timestamp = await self._fetch_last_commit_timestamp(
```
Comment (seanstory) [inline connectors/sources/github.py:1290]: Should we be fetching the last_commit_timestamp for the markdown files? I see that we weren't before when type was `BLOB`, but I don't understand why.

**Review by seanstory — COMMENTED** [review]
two clarity-seeking comments. But largely looks good.

File: connectors/sources/github.py [hunk 2]
```diff
@@ -1287,39 +1288,32 @@ async def _fetch_files(self, repo_name, default_branch):
...
-                    repo_name=repo_name, path=repo_object["path"]
-                )
                 if repo_object["type"] == BLOB:
```
[the remark this answers is not in the document]
Comment (abhishekjoshi-crest) [reply connectors/sources/github.py:1291]: In the response, we are getting two types of documents one is `blob` and the other is `tree`. For the files, we are getting `blob` type and for the folders we are getting `tree` type. So, markdown files fall under `blob` type.

File: connectors/sources/github.py (hunk 1 above)
Comment (abhishekjoshi-crest) [reply connectors/sources/github.py:1290]: Earlier, we were fetching `last_commit_timestamp` for all the documents as we were indexing all the files. but now we are indexing markdown files only, so we are fetching `last_commit_timestamp` for markdown files only in L1299.

Comment (seanstory) [reply connectors/sources/github.py:1290]: omg Idk how I missed that. Thanks.

**Pull request merged**